include/ddc/chunk_span.hpp

```diff
@@ -260,30 +262,58 @@ class ChunkSpan<ElementType, DiscreteDomain<DDims...>, LayoutStridedPolicy, Memo
     KOKKOS_FUNCTION constexpr auto operator[](
             DiscreteElement<QueryDDims...> const& slice_spec) const
     {
-        auto subview = std::experimental::
-                submdspan(allocation_mdspan(), get_slicer_for<DDims>(slice_spec)...);
+        auto subview = Kokkos::submdspan(allocation_mdspan(), get_slicer_for<DDims>(slice_spec)...);
+        using layout_type = typename decltype(subview)::layout_type;
+        using extents_type = typename decltype(subview)::extents_type;
         using detail::TypeSeq;
         using OutTypeSeqDDims = type_seq_remove_t<TypeSeq<DDims...>, TypeSeq<QueryDDims...>>;
         using OutDDom = detail::convert_type_seq_to_discrete_domain_t<OutTypeSeqDDims>;
-        return ChunkSpan<
-                ElementType,
-                OutDDom,
-                typename decltype(subview)::layout_type,
-                memory_space>(subview, OutDDom(this->m_domain));
+        if constexpr (
+                !std::is_same_v<layout_type, Kokkos::Experimental::layout_left_padded<>>
+                && !std::is_same_v<layout_type, Kokkos::Experimental::layout_right_padded<>>) {
+            return ChunkSpan<
+                    ElementType,
+                    OutDDom,
+                    layout_type,
+                    memory_space>(subview, OutDDom(this->m_domain));
+        } else {
+            Kokkos::layout_stride::mapping<extents_type> const mapping_stride(subview.mapping());
+            Kokkos::mdspan<ElementType, extents_type, Kokkos::layout_stride> const
+                    a(subview.data_handle(), mapping_stride);
+            return ChunkSpan<
+                    ElementType,
+                    OutDDom,
+                    Kokkos::layout_stride,
+                    memory_space>(a, OutDDom(this->m_domain));
+        }
```

Comment:
```suggestion
```suggestion
        if constexpr (std::is_same_v<layout_type, Kokkos::Experimental::layout_left_padded<>> || 
           std::is_same_v<layout_type, Kokkos::Experimental::layout_right_padded<>>) {
           Kokkos::layout_stride::mapping<extents_type> const mapping_stride(subview.mapping());
             Kokkos::mdspan<ElementType, extents_type, Kokkos::layout_stride> const
                     a(subview.data_handle(), mapping_stride);
             return ChunkSpan<
                     ElementType,
                     OutDDom,
                     Kokkos::layout_stride,
                     memory_space>(a, OutDDom(this->m_domain));
        } else {
          return ChunkSpan<
                     ElementType,
                     OutDDom,
                     layout_type,
                     memory_space>(subview, OutDDom(this->m_domain));
        }
```
```

Reply:
Yes, the idea is that `( (not A) and (not B) ) == not (A or B)`. It is simpler and more readable to use check if `(A or B)`. The if/else branches are inverted because it checks `(A or B)` rather than `not (A or B)`

Reply:
fixed in 49c89244b2b6db06209098c6cf4c5a9676f8b5ae